src/timeMachine/components/dynamicFluxFunctionsToolbar/FluxFunctionsToolbar.tsx

```diff
@@ -8,56 +9,105 @@ import {DapperScrollbars} from '@influxdata/clockface'
 import ErrorBoundary from 'src/shared/components/ErrorBoundary'
 import ToolbarFunction from 'src/timeMachine/components/dynamicFluxFunctionsToolbar/ToolbarFunction'
 
-// Constants
-import {FLUX_FUNCTIONS} from 'src/shared/constants/fluxFunctions'
+import {SpinnerContainer, TechnoSpinner} from '@influxdata/clockface'
+
+import {Fluxdocs} from 'src/client/fluxdocsdRoutes'
+import {getFluxPackages} from 'src/timeMachine/actions/scriptEditorThunks'
 
 // Types
-import {FluxToolbarFunction} from 'src/types'
+import {RemoteDataState} from 'src/types'
+import {AppState} from 'src/types'
 
 interface OwnProps {
-  onInsertFluxFunction: (func: FluxToolbarFunction) => void
+  onInsertFluxFunction: (func) => void
+}
+
+interface DispatchProps {
+  getFluxPackages: () => void
 }
 
-const DynamicFluxFunctionsToolbar: FC<OwnProps> = (props: OwnProps) => {
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = ReduxProps & OwnProps & DispatchProps
+
+const DynamicFluxFunctionsToolbar: FC<Props> = (props: Props) => {
   const [searchTerm, setSearchTerm] = useState('')
+  const [fluxServiceError, setFluxServiceError] = useState<RemoteDataState>(
+    RemoteDataState.NotStarted
+  )
 
   const handleSearch = (searchTerm: string): void => {
     setSearchTerm(searchTerm)
   }
 
+  useEffect(() => {
+    const getFluxFuncs = async () => {
+      try {
+        setFluxServiceError(RemoteDataState.Loading)
+
+        if (props.fluxFunctions.length === 0) {
+          await props.getFluxPackages()
+          setFluxServiceError(RemoteDataState.Done)
+        }
+        setFluxServiceError(RemoteDataState.Done)
+      } catch (err) {
+        console.error(err)
+        setFluxServiceError(RemoteDataState.Error)
+      }
+    }
+    getFluxFuncs()
+  }, [])
```

Comment:
Add comment at the end of this line:
```
// eslint-disable-line react-hooks/exhaustive-deps
```

Reply:
with the way we are using useEffect hook here, I'm pretty confident we won't need to add anything in the dependency array. So, it makes sense to disable the `react-hooks/exhaustive-deps` rule